addons/mail/static/src/js/chat_manager.js

```diff
@@ -506,6 +507,7 @@ function on_toggle_star_notification (data) {
             chat_manager.bus.trigger('update_message', message);
         }
     });
+    chat_manager.bus.trigger('update_starred');
```

Comment:
you should also update starred_counter here.  Either decrement or increment, depending on message.is_starred